cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/io/cdap/cdap/datapipeline/draft/DraftStore.java

```diff
@@ -0,0 +1,302 @@
+          right = o2.getString(sortField);
+          break;
+        case INTEGER:
+          left = o1.getInteger(sortField);
+          right = o2.getInteger(sortField);
+          break;
+        case LONG:
+          left = o1.getLong(sortField);
+          right = o2.getLong(sortField);
+          break;
+        case FLOAT:
+          left = o1.getFloat(sortField);
+          right = o2.getFloat(sortField);
+          break;
+        case DOUBLE:
+          left = o1.getDouble(sortField);
+          right = o2.getDouble(sortField);
+          break;
+        case BYTES:
+          throw new IllegalArgumentException("Cannot sort field '%s' because it is of type BYTES.");
+      }
+      if (left == null && right == null) {
+        return 0;
+      }
+      if (left == null || right == null) {
+        result = left == null ? -1 : 1;
+      } else {
+        result = ((Comparable<Object>) left).compareTo(right);
+      }
+
+      if (sortRequest.getOrder() == SortRequest.SortOrder.ASC) {
+        return result;
+      }
+      return -1 * result;
+    });
+
+    return rows;
+  }
+
+  /**
+   * Fetch a given draft if it exists
+   *
+   * @param id {@link DraftId} that is used to uniquely identify a draft
+   * @return an {@link Optional<Draft>} representing the requested draft
+   */
+  public Optional<Draft> getDraft(DraftId id) throws TransactionException {
+    AtomicReference<Optional<Draft>> result = new AtomicReference<>();
+    transactionRunner.run(context -> {
+      StructuredTable table = context.getTable(TABLE_ID);
+      Optional<StructuredRow> row = table.read(getKey(id));
+      result.set(row.map(this::fromRow));
+    });
+    return result.get();
+  }
+
+  /**
+   * Delete the given draft. This is a no-op if the draft does not exist
+   *
+   * @param id {@link DraftId} that is used to uniquely identify a draft
+   */
+  public void deleteDraft(DraftId id) throws TransactionException {
+    transactionRunner.run(context -> {
+      StructuredTable table = context.getTable(TABLE_ID);
+      table.delete(getKey(id));
+    });
+  }
+
+  /**
+   * Create/update the given draft
+   *
+   * @param id {@link DraftId} that is used to uniquely identify a draft
+   * @param request {@link DraftRequest} that contains the rest of the draft data
+   * @throws TransactionException
+   */
+  public <T extends ETLConfig> void writeDraft(DraftId id, DraftRequest<T> request) throws TransactionException {
+    Optional<Draft> existing = getDraft(id);
+    long now = System.currentTimeMillis();
+    long createTime = existing.map(Draft::getCreatedTimeMillis).orElse(now);
+
+    Draft draft = new Draft(request.getConfig(), request.getName(), request.getDescription(), request.getArtifact(),
+                            id.getId(), createTime, now);
+    transactionRunner.run(context -> {
+//      StructuredTable table = context.getTable(TABLE_ID);
+            StructuredTable table = context.getTable(new StructuredTableId("TESTST"));
```

Comment:
Is this correct?

Reply:
Whoops, that was just for testing. Shouldnt have committed that.